util/CosmosHelper.js

```diff
@@ -1,25 +1,34 @@
 import BigNumber from 'bignumber.js';
 import {
-  COSMOS_CHAIN_ID,
   COSMOS_DENOM,
+  BASIC_TRANSFER_GAS,
+  EXTERNAL_URL,
 } from '@/constant';
 import { timeout } from '@/util/misc';
+import { queryTxInclusion } from '@/util/cosmos/misc';
+import {
+  CosmosClient, SigningCosmosClient,
+  GasPrice,
+} from '@cosmjs/launchpad';
 
-export const DEFAULT_GAS_PRICE = [{ amount: 1000, denom: 'nanolike' }];
-export const DEFAULT_GAS_PRICE_NUMBER = DEFAULT_GAS_PRICE[0].amount;
+export const DEFAULT_GAS_PRICE = [{ amount: '1000', denom: 'nanolike' }];
+export const DEFAULT_GAS_PRICE_NUMBER = parseInt(DEFAULT_GAS_PRICE[0].amount, 10);
 export const DEFAULT_ISCN_GAS_PRICE = [{ amount: 0, denom: 'nanolike' }];
-const COSMOS_RESTFUL_API = '/api/cosmos/lcd';
+const COSMOS_RESTFUL_API = `${EXTERNAL_URL}/api/cosmos/lcd`;
+
+let cosmosClient;
+let signingCosmosClient;
 
-let Cosmos;
-let api;
+async function initCosmosClient() {
+  if (cosmosClient) return;
+  cosmosClient = new CosmosClient(COSMOS_RESTFUL_API);
+}
```

Comment:
which part of this function requires it to be `async` ?

Reply:
true. changed to only function.